docs/architecture.rst

```diff
@@ -367,25 +367,32 @@ usage before and after the instantiation. The memory usage is then averaged over
 synapses instantiated. The script used to perform this operation `synstat.py` is available for the user
 and is archived in this repo in the `_benchmarks` folder.
 
-Having these pre-computed values allows us to simply count the amount of synapses of each type
-and multiply it by the corresponding memory usage value.
+Having these pre-computed values, we just need to estimate the amount of synapses of each type
+and multiply it by the corresponding memory usage value. Even in this case we have adopted a
+sample-based approach. First of all, we filter out any gids that have already been
+counted in a previous run. This is done to avoid double-counting synapses.
+Then we load all the cells of the circuit in progressively bigger blocks in order to avoid any OOM issues
+while keeping good performances.
+
+For each metype and block, we estimate the number of synapses by sampling a subset (100) of the gids
+and counting the number of synapses for each gid in the sample. The counts are then extrapolated
+linearly to the entire population of gids for that metype.
+
+We used a sophisticated sampling strategy to optimize performance. It processes the gids in blocks,
+starting with small blocks (5000) and gradually increasing the block size using a geometric progression.
+Within each block, it samples a fixed number of gids (always 100).
+This strategy takes advantage of data locality and allows us to handle very large sets of gids.
```

Comment:
I think the idea is getting a bit repeated. We could maybe condense this and go straight to the point. Also `counts are then extrapolated linearly to the entire population` is not entirely true. They are extrapolated linearly only inside the block to work with skewed data as we have recently changed.
What about. ?
The parameters of the sampling are as follows:
 - Blocks start with length of 5000, increasing at a rate of 10%
 - Count synapses for the first 100 cells of the block (take advantage of data locality)
 - Extrapolate for the whole block and add to the global metype estimate.